### DISCUSSION TITLE
Error exectuting first Poisson example under Windows/conda

### DISCUSSION THREAD

**Discussion by maurosgroi** [kinnala/scikit-fem#996] [Q&A]
Dear Developers,
my name is Mauro Sgroi and I am evaluating the use of Scikit-Fem to teach FEM in a course.
I can run the examples on Linux, but I have to use conda on windows since the major parts of my students have Windows based PC.
Running the first Poisson example:

from skfem import *
from skfem.helpers import dot, grad
from pathlib import Path

from skfem import *
from skfem.models.poisson import laplace, unit_load
from skfem.io.json import from_file

import numpy as np

m = MeshTri.init_circle(4)

basis = Basis(m, ElementTriP2())

A = asm(laplace, basis)
b = asm(unit_load, basis)

x = solve(*condense(A, b, D=basis.get_dofs()))

area = sum(b)
k = b @ x / area**2
k1, = basis.probes(np.zeros((2, 1))) @ x / area

def visualize():
    from skfem.visuals.matplotlib import plot
    return plot(basis, x, shading='gouraud', colorbar=True)

if __name__ == '__main__':
    print('area = {:.4f} (exact = {:.4f})'.format(area, np.pi))
    print('k = {:.5f} (exact = 1/8/pi = {:.5f})'.format(k, 1/np.pi/8))
    print("k' = {:.5f} (exact = 1/4/pi = {:.5f})".format(k1, 1/np.pi/4))
    visualize().show()

I get the error:

runfile('C:/Users/Utente/Documents/Mauro/Python/skfem/poisson1.py', wdir='C:/Users/Utente/Documents/Mauro/Python/skfem')
Traceback (most recent call last):

  File "C:\Users\Utente\Documents\Mauro\Python\skfem\poisson1.py", line 15, in <module>
    A = asm(laplace, basis)

  File "C:\Users\Utente\anaconda3\lib\site-packages\skfem\assembly\__init__.py", line 62, in asm
    return form.assemble(*args, **kwargs)

  File "C:\Users\Utente\anaconda3\lib\site-packages\skfem\assembly\form\bilinear_form.py", line 56, in assemble
    **ubasis.default_parameters(),

  File "C:\Users\Utente\anaconda3\lib\site-packages\skfem\assembly\basis\basis.py", line 162, in default_parameters
    raise NotImplementedError("Default parameters not implemented.")

NotImplementedError: Default parameters not implemented.

How can I solve this problem?

Thanks a lot and best regards,
Mauro Sgroi.

**Comment by kinnala** [comment]
Thanks for the report. We currently have no active contributors using Windows so we are only relying on Github Actions to do its job.

However, I think in this case the issue is that you have installed an old version of scikit-fem using conda where `Basis` means a different thing than in the most recent version of the code. The maintainer of the conda-forge package has not been actively updating the recipe. As you can see here the conda-forge package is almost 2 years old: https://anaconda.org/conda-forge/scikit-fem

If I remember correctly, in the old version you'd need to replace `Basis` by `InteriorBasis`.

However, I suggest you use a more recent version of the code since the documentation has been already updated to match the most recent versions. The official way of installing scikit-fem is through pip: `pip install scikit-fem[all]`. You can do this also in conda environment by first making sure you have installed `pip` through conda.

**Reply by kinnala** [reply]
As you can see here I've tried some interaction with the maintainer of the conda-forge scikit-fem package about 1 year 4 months ago and was left unanswered: https://github.com/conda-forge/scikit-fem-feedstock/pull/17

**Reply by kinnala** [reply]
I'm now working on to get the feedstock updated although it won't become the official way to install the package: https://github.com/conda-forge/scikit-fem-feedstock/issues/18#issuecomment-1355154942

**Comment by maurosgroi** [comment]
Dear Kinnala,
thanks a lot for the explanation.
It works now.
Best regards,
Mauro.

**Reply by kinnala** [reply]
Great! Let me know if you have any further questions or improvement ideas.

**Comment by maurosgroi** [comment]
For sure.
Best regards,
Mauro.
